### PR TITLE
Compatible with scribus 1.5.7

### PR DESCRIPTION
**Pull request by ta-ichihashi**
I got minimum fix to work it on scribus 1.5.7.

### CODE REVIEW

File: ScribusGeneratorBackend.py [hunk 1]
```diff
@@ -214,8 +214,9 @@ def parse_data(self):
         if extension == '.csv':
             # .. from CSV file
             data = self.load_csv(data_file)
+            logging.debug(data)
 
-            if len(data) <= 1:
```
Comment (berteh) [inline ScribusGeneratorBackend.py:218]: why this change ? if data.len is 1 then there's only titles, not values to substitute > drop.

File: ScribusGeneratorBackend.py [hunk 2]
```diff
@@ -417,6 +419,14 @@ def generate_templates(self, root, data: list) -> list:
 
         # Clean & write single SLA file (merge-mode only)
         if merge_mode:
+            output_element = ET.fromstring(output)
```
Comment (berteh) [inline ScribusGeneratorBackend.py:422]: i dont quite understand the purpose of these next 4lines. What problem do you fix here ?

Comment (berteh) [reply ScribusGeneratorBackend.py:422]: i.e. 422 to 425.

426 is a nice addition I forgot, thanks.

File: ScribusGeneratorBackend.py (hunk 1 above)
Comment (berteh) [inline ScribusGeneratorBackend.py:218]: why this change ? if data.len is 1 then there's only titles, no values to substitute > skip.

**Review by berteh — CHANGES_REQUESTED** [review]
thanks for the PR. kindly have a look at my few comments.

File: ScribusGeneratorBackend.py (hunk 1 above)
Comment (ta-ichihashi) [reply ScribusGeneratorBackend.py:218]: The variable 'data' is 'OrederdDict' list types which has title and data row pair.
I was faced getting this error when loading csv file which has only one record.

File: ScribusGeneratorBackend.py [hunk 3]
```diff
@@ -400,7 +401,8 @@ def generate_templates(self, root, data: list) -> list:
                         version
                     )
 
-                    document_element.extend(shifted_elements)
+                    if index != min(records_in_document, data_count):
```
[the remark this answers is not in the document]
Comment (ta-ichihashi) [reply ScribusGeneratorBackend.py:404]: Sorry, I had not know "%SG_NEXT-RECORD%" feature.
I will learn it and consider another way not to be printed twice on first document.

File: ScribusGeneratorBackend.py (hunk 2 above)
Comment (ta-ichihashi) [reply ScribusGeneratorBackend.py:422]: The reason of this 4-lines addition was to fix the issue missing following tags.

`<SCRIBUSUTF8NEW Version="1.5.7">.... </SCRIBUSUTF8NEW>`

Sorry, right now, I am not sure whether this 4-lines effective. This issue was not appeared when I tried again with commented this codes out.

**Comment by ta-ichihashi** [comment]
I fixed for all comment and tested with using "%SG_NEXT-RECORD%". Then, I found new issue getting "IndexError" by undefined items for last page. Also fixed it.
Please review again.

Thanks.

File: ScribusGeneratorBackend.py [hunk 4]
```diff
@@ -401,7 +401,7 @@ def generate_templates(self, root, data: list) -> list:
                         version
                     )
 
-                    if index != min(records_in_document, data_count):
+                    if index > records_in_document:
```
Comment (berteh) [inline ScribusGeneratorBackend.py:404]: Thanks for the update.
This might require a test with a template that uses multiple data entries (i.e. a few %SG_NEXT-RECORD%, eg 5) and a data file that has some entries but LESS that needed to fill the whole document (eg 3).

I wanted to write a basic test suite against regression bugs for a long time... maybe now it would have been useful :/

**Review by berteh — COMMENTED** [review]
thanks for the update. I'll just run a few bordercase tests and then approve.
